proxy/src/bind.rs

```diff
@@ -67,7 +67,7 @@ pub enum Binding<B: tower_h2::Body + 'static> {
 /// that each host receives its own connection.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub enum Protocol {
-    Http1(Host),
+    Http1 { host: Host, was_absolute_form: bool },
```

Comment:
If I understand correctly, this change moves this information into the router's `Key`. Is that correct?  Why?

Can you update the comment to indicate what this field means?

Reply:
The change to hyper doesn't sound good to me, if it results in us needing to make changes like this.

Reply:
@briansmith I'm also curious to know what motivated this change in Hyper. I suspect @seanmonstar might be able to offer some insight?

Reply:
Thanks for the context, @seanmonstar, I suspected it was something along those lines.

Reply:
@briansmith, have you had a chance to look at @seanmonstar's comment yet? I'd love to get your opinion before we move forwards with this PR.

Reply:
> @briansmith, have you had a chance to look at @seanmonstar's comment yet? I'd love to get your opinion before we move forwards with this PR.

If everybody else is fine with it, then go ahead.